### PR TITLE
wgnord: init at 0.1.10

### LINKED ISSUE #201573: Package Request: wgnord (nordvpn wireguard script)
**Project description**
I've tried packaging [this script](https://github.com/phirecc/wgnord) myself but as a complete newbie to Nix, I didn't get anywhere with it. It is a convenient way to connect to Nord VPN without installing any closed source packages. It seems the nordlynx "protocol" is essentially vanilla wireguard with a special private key retrieved through an API.

I have seen issue #132082 but I think this could be a good free alternative for anyone wanting to connect to Nord VPN with wireguard.

**Metadata**

* homepage URL: https://github.com/phirecc/wgnord
* source URL: https://github.com/phirecc/wgnord
* license: not stated
* platforms: linux

### PR DESCRIPTION
**Pull request by urandom2**
###### Description of changes

Fixes #201573

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/wgnord/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/networking/wgnord/default.nix:27]:
```suggestion
  postPatch = ''
```

File: pkgs/applications/networking/wgnord/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/networking/wgnord/default.nix:32]:
```suggestion

  dontBuild = true;

```

File: pkgs/applications/networking/wgnord/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/networking/wgnord/default.nix:41]: same for those two

File: pkgs/applications/networking/wgnord/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/networking/wgnord/default.nix:38]: We don't really need this on nixos. You can only create a system without coreutils by using lib.mkForce on environment.systemPackages and then most things will break https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/config/system-path.nix#L15

Comment (urandom2) [reply pkgs/applications/networking/wgnord/default.nix:38]: sure, but since we are targeting more than just nixos, it is helpful to pin away from system coreutils? this always bites me on darwin.

Comment (SuperSandro2000) [reply pkgs/applications/networking/wgnord/default.nix:40]: I would suggest to globally install updates of all the gnu tools instead of hardcoding them into way to many packages.

File: pkgs/applications/networking/wgnord/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/networking/wgnord/default.nix:28]:
```suggestion
    substituteInPlace wgnord \
```

File: pkgs/applications/networking/wgnord/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/networking/wgnord/default.nix:36]:
```suggestion
    install -Dm755 wgnord -t $out/bin/
```

File: pkgs/applications/networking/wgnord/default.nix
Comment (urandom2) [reply pkgs/applications/networking/wgnord/default.nix:40]: if this is a hard requirement, I am happy to apply it, but this has already caused unpredictable behaviour on darwin and bsd systems for me

Comment (SuperSandro2000) [reply pkgs/applications/networking/wgnord/default.nix:40]: It is unreasonable amounts of work to add those packages to all packages that require them. The simpler and better to maintain solution is to add those packages on systems that do not ship them.

> I would also ask what users on those oses who need the builtin versions for system scripting, and gnu versions for nixpkgs do?

PATH is only expanded in the user session. It would be even more confusing to have different core programs in different programs.

---

Short: I am against hardcoding such core programs into packages, if other distros require them, add them to the nixos installation.

Comment (urandom2) [reply pkgs/applications/networking/wgnord/default.nix:40]: > It is unreasonable amounts of work to add those packages to all packages that require them.

If this is so much work, why do we do it this way for every shared object?

> The simpler and better to maintain solution is to add those packages on systems that do not ship them.

I agree it is simpler for nixpkgs, but it leads to heisen bugs and "works on my machine" issuues. I hit this exact issue with find and getopt recently, and it was incredibly confusing to the user. Furthermore, it undermined the "nix workes the same everywhere" promise they had been sold.

> PATH is only expanded in the user session. It would be even more confusing to have different core programs in different programs.

This is exactly requirement for how my corporate macbook works. There are system daemons and scripts that the company maintains, that use the built-in versions of find and other system tools. Then I have nix which runs things that expect both proprietary system utilities (launched) and nix dependencies (gnu find, not bsd find). I would expect being explicit is the only way to make this clear?

> Short: I am against hardcoding such core programs into packages, if other distros require them, add them to the nixos installation.

Can you elaborate on what you mean? Nixos is the one that works ootb, so I am unclear what "add[ing] them to nixos installation" does to resolve the issue, or do you mean the "nix install" should setup PATH to contain everything in stdenv?

Comment (abathur) [reply pkgs/applications/networking/wgnord/default.nix:40]: > should we be pushing maintainers to patch relative and bare paths to absolute paths into all binary executions?

This is, at least for Bash/POSIX Shell, what https://github.com/abathur/resholve exists to do.

Your call, as maintainer/packager, but I made a quick example atop your PR:

https://github.com/abathur/nixpkgs/commit/d6030d4be4a6c8fb084f617d9991b5211405596e

<details><summary>And for reference, the ~resholved script:</summary>

```console
$ cat /nix/store/h76vnv7h9lv3n0v10qfiicx81yb4n3vv-wgnord-0.1.10/bin/wgnord 
Fri Jan 06 2023 20:38:57 --> 
#!/nix/store/4xw8n979xpivdc46a9ndcvyhwgif00hz-bash-5.1-p16/bin/sh
alias query="/nix/store/m2h1p50yvcq5j9b3hkrwqnmrr9pbkzpz-curl-7.86.0-bin/bin/curl -s -H 'User-Agent: NordApp Linux 3.12.0 Linux 5.4.0-58-generic'"
alias print="printf \"\033[32m%s\n\033[0m\""
alias print_error="printf \"\033[31m%s\n\033[0m\""
host="api.nordvpn.com"
conf_dir=/var/lib/wgnord
check_root() {
	if [ "$(/nix/store/h8gvq6r4hgpa71h44dmg9qfx03mj81sv-coreutils-9.1/bin/id -u)" -ne 0 ]; then
		echo "This script must be run as root"
		exit 1
	fi
}
login() {
	[ -z "$1" ] && help
	pass="$2"
	if [ -z "$pass" ]; then
		printf "Enter password: "
		/nix/store/h8gvq6r4hgpa71h44dmg9qfx03mj81sv-coreutils-9.1/bin/stty -echo
		read pass
		/nix/store/h8gvq6r4hgpa71h44dmg9qfx03mj81sv-coreutils-9.1/bin/stty echo
		echo
	fi
	print "Logging in..."
	umask 027
	/nix/store/vw9lf2d0cfw92dsascyv7q1l5w260r2g-jq-1.6-bin/bin/jq -njc "{username: \"$1\", password: \"$(echo "$pass" | /nix/store/5dv5cq1lwvsijr9p16p2kp79g1dbajk3-gnused-4.8/bin/sed 's/"/\\"/g')\"}" > $conf_dir/account.json
	get_token
	get_credentials
}
get_token() {
	print "Getting new access token..."
	query "https://$host/v1/users/tokens" -d "@$conf_dir/account.json" -H "Content-Type: application/json" > $conf_dir/token.json
	err="$(/nix/store/vw9lf2d0cfw92dsascyv7q1l5w260r2g-jq-1.6-bin/bin/jq -er '.errors.message' $conf_dir/token.json)" && { print_error "Error getting token: $err"; exit 1; }
}
get_credentials() {
	print "Getting server credentials..."
	token_date=$(/nix/store/h8gvq6r4hgpa71h44dmg9qfx03mj81sv-coreutils-9.1/bin/date -d "$(/nix/store/vw9lf2d0cfw92dsascyv7q1l5w260r2g-jq-1.6-bin/bin/jq -r '.expires_at' $conf_dir/token.json)" --utc +%s)
	current_date=$(/nix/store/h8gvq6r4hgpa71h44dmg9qfx03mj81sv-coreutils-9.1/bin/date --utc +%s)
	[ "$current_date" -ge $((token_date-60)) ] && get_token
	auth_token="$(printf "%s" "token:$(/nix/store/vw9lf2d0cfw92dsascyv7q1l5w260r2g-jq-1.6-bin/bin/jq -j '.token' $conf_dir/token.json)" | /nix/store/h8gvq6r4hgpa71h44dmg9qfx03mj81sv-coreutils-9.1/bin/base64 -w 0)"
	query "https://$host/v1/users/services/credentials" -H "Content-Type: application/json" -H "Authorization: Basic $auth_token" > $conf_dir/credentials.json
	err="$(/nix/store/vw9lf2d0cfw92dsascyv7q1l5w260r2g-jq-1.6-bin/bin/jq -er '.errors.message' $conf_dir/credentials.json)" && { print_error "Error getting credentials: $err"; exit 1; }
}
get_country_code() {
    buf="$(/nix/store/bcvccw6y9bfil6xrl5j7psza7hnd16ry-gnugrep-3.7/bin/grep -i "^$1	" -m 1 /nix/store/8pc71rrw00qksrz9l6x313i4xjlw0zp4-wgnord-unresholved-0.1.10/share/countries_iso31662.txt | /nix/store/h8gvq6r4hgpa71h44dmg9qfx03mj81sv-coreutils-9.1/bin/cut -d "	" -f 2)"
    [ -z "$buf" ] && buf="$(/nix/store/bcvccw6y9bfil6xrl5j7psza7hnd16ry-gnugrep-3.7/bin/grep -i "$1" -m 1 /nix/store/8pc71rrw00qksrz9l6x313i4xjlw0zp4-wgnord-unresholved-0.1.10/share/countries.txt | /nix/store/h8gvq6r4hgpa71h44dmg9qfx03mj81sv-coreutils-9.1/bin/cut -d "	" -f 2)"
    echo "$buf"
}
connect() {
	out_file=/etc/wireguard/wgnord.conf
	OPTIND=2
	while getopts "fno:" o; do case "$o" in
		f) force=1 ;;
		n) dont_act=1 ;;
		o) out_file="$OPTARG" ;;
		*) exit 1 ;;
	esac done
	[ -z "$1" ] && help
	print "Finding best server..."
	if [ $force ] || [ ! -f $conf_dir/coords.json ] && [ ! $dont_act ]; then
		is_connected && disconnect
		insights="$(query "https://$host/v1/helpers/ips/insights")"
		longitude="$(echo "$insights" | /nix/store/vw9lf2d0cfw92dsascyv7q1l5w260r2g-jq-1.6-bin/bin/jq -j '.longitude')"
		latitude="$(echo "$insights" | /nix/store/vw9lf2d0cfw92dsascyv7q1l5w260r2g-jq-1.6-bin/bin/jq -j '.latitude')"
		/nix/store/vw9lf2d0cfw92dsascyv7q1l5w260r2g-jq-1.6-bin/bin/jq -njc "{longitude: $longitude, latitude: $latitude}" > $conf_dir/coords.json
	else
		longitude="$(/nix/store/vw9lf2d0cfw92dsascyv7q1l5w260r2g-jq-1.6-bin/bin/jq -er '.longitude' $conf_dir/coords.json)"
		latitude="$(/nix/store/vw9lf2d0cfw92dsascyv7q1l5w260r2g-jq-1.6-bin/bin/jq -er '.latitude' $conf_dir/coords.json)"
	fi
	country_code="$(get_country_code "$1")"
	recommendations="$(query "https://$host/v1/servers/recommendations?limit=20&filters%5Bservers.status%5D=online&filters%5Bservers_technologies%5D=35&filters%5Bservers_technologies%5D%5Bpivot%5D%5Bstatus%5D=online&fields%5Bservers.id%5D&fields%5Bservers.name%5D&fields%5Bservers.hostname%5D&fields%5Bservers.station%5D&fields%5Bservers.status%5D&fields%5Bservers.load%5D&fields%5Bservers.created_at%5D&fields%5Bservers.groups.id%5D&fields%5Bservers.groups.title%5D&fields%5Bservers.technologies.id%5D&fields%5Bservers.technologies.metadata%5D&fields%5Bservers.technologies.pivot.status%5D&fields%5Bservers.specifications.identifier%5D&fields%5Bservers.specifications.values.value%5D&fields%5Bservers.locations.country.name%5D&fields%5Bservers.locations.country.code%5D&fields%5Bservers.locations.country.city.name%5D&fields%5Bservers.locations.country.city.latitude%5D&fields%5Bservers.locations.country.city.longitude%5D&coordinates%5Blongitude%5D=$longitude&coordinates%5Blatitude%5D=$latitude&fields%5Bservers.ips%5D&filters%5Bcountry_id%5D=$country_code")"
	server_name="$(echo "$recommendations" | /nix/store/vw9lf2d0cfw92dsascyv7q1l5w260r2g-jq-1.6-bin/bin/jq -j '.[0] | .name')"
	server_hostname="$(echo "$recommendations" | /nix/store/vw9lf2d0cfw92dsascyv7q1l5w260r2g-jq-1.6-bin/bin/jq -j '.[0] | .hostname')"
	server_ip="$(echo "$recommendations" | /nix/store/vw9lf2d0cfw92dsascyv7q1l5w260r2g-jq-1.6-bin/bin/jq -j '.[0] | .ips[0].ip.ip')"
	server_pubkey="$(echo "$recommendations" | /nix/store/vw9lf2d0cfw92dsascyv7q1l5w260r2g-jq-1.6-bin/bin/jq -j '.[0] | .technologies[] | select(.metadata[0].name == "public_key") | .metadata[0].value')"
	privkey="$(/nix/store/vw9lf2d0cfw92dsascyv7q1l5w260r2g-jq-1.6-bin/bin/jq -j '.nordlynx_private_key' $conf_dir/credentials.json)"
	/nix/store/5dv5cq1lwvsijr9p16p2kp79g1dbajk3-gnused-4.8/bin/sed -e "s|PRIVKEY|$privkey|" -e "s|SERVER_PUBKEY|$server_pubkey|" -e "s|SERVER_IP|$server_ip|" $conf_dir/template.conf > "$out_file"
	if [ ! $dont_act ]; then
		print "Connecting to $server_hostname ($server_name)..."
		if is_connected; then
			/nix/store/asdp7lnwclv2s08ajm0v6srmwc1wpv09-wireguard-tools-1.0.20210914/bin/wg-quick strip wgnord | /nix/store/asdp7lnwclv2s08ajm0v6srmwc1wpv09-wireguard-tools-1.0.20210914/bin/wg setconf wgnord /dev/stdin
			/nix/store/asdp7lnwclv2s08ajm0v6srmwc1wpv09-wireguard-tools-1.0.20210914/bin/wg set wgnord fwmark 0xca6c
		else
			/nix/store/asdp7lnwclv2s08ajm0v6srmwc1wpv09-wireguard-tools-1.0.20210914/bin/wg-quick up wgnord
		fi && print "Connected successfully!"
	fi
}
disconnect() {
	print "Disconnecting..."
	/nix/store/asdp7lnwclv2s08ajm0v6srmwc1wpv09-wireguard-tools-1.0.20210914/bin/wg-quick down wgnord
}
is_connected() {
	/nix/store/vs4303zaqb1grq6kgf2z7wsjzlrz71xy-iproute2-6.0.0/bin/ip link show wgnord > /dev/null 2>&1
	return
}
help() { /nix/store/h8gvq6r4hgpa71h44dmg9qfx03mj81sv-coreutils-9.1/bin/cat << EOF
Usage: wgnord [ l(ogin) | c(onnect) | d(isconnect) | g(et_credentials) ]

login:
    wgnord l "john.smith@example.com" "verysecurepassword"
    If the password was omitted it will be read from stdin
connect:
    wgnord c france
    -f            Refresh cached longitude/latitude
    -n            Don't connect
    -o out.conf   Write config to different file
disconnect:
    wgnord d
get_credentials:
    wgnord g
    This typically isn't needed. You can try running this if the connection fails

wgnord's files are in $conf_dir, edit template.conf to change Wireguard options
EOF
	exit
}

case $1 in
	l|login) check_root; login "$2" "$3" ;;
	c|connect) check_root; shift; connect "$@" ;;
	d|disconnect) check_root; disconnect ;;
	g|get_credentials) check_root; get_credentials ;;
	*) help ;;
esac

### resholve directives (auto-generated) ## format_version: 3
# resholve: fix aliases
# resholve: keep /nix/store/5dv5cq1lwvsijr9p16p2kp79g1dbajk3-gnused-4.8/bin/sed
# resholve: keep /nix/store/asdp7lnwclv2s08ajm0v6srmwc1wpv09-wireguard-tools-1.0.20210914/bin/wg
# resholve: keep /nix/store/asdp7lnwclv2s08ajm0v6srmwc1wpv09-wireguard-tools-1.0.20210914/bin/wg-quick
# resholve: keep /nix/store/bcvccw6y9bfil6xrl5j7psza7hnd16ry-gnugrep-3.7/bin/grep
# resholve: keep /nix/store/h8gvq6r4hgpa71h44dmg9qfx03mj81sv-coreutils-9.1/bin/base64
# resholve: keep /nix/store/h8gvq6r4hgpa71h44dmg9qfx03mj81sv-coreutils-9.1/bin/cat
# resholve: keep /nix/store/h8gvq6r4hgpa71h44dmg9qfx03mj81sv-coreutils-9.1/bin/cut
# resholve: keep /nix/store/h8gvq6r4hgpa71h44dmg9qfx03mj81sv-coreutils-9.1/bin/date
# resholve: keep /nix/store/h8gvq6r4hgpa71h44dmg9qfx03mj81sv-coreutils-9.1/bin/id
# resholve: keep /nix/store/h8gvq6r4hgpa71h44dmg9qfx03mj81sv-coreutils-9.1/bin/stty
# resholve: keep /nix/store/m2h1p50yvcq5j9b3hkrwqnmrr9pbkzpz-curl-7.86.0-bin/bin/curl
# resholve: keep /nix/store/vs4303zaqb1grq6kgf2z7wsjzlrz71xy-iproute2-6.0.0/bin/ip
# resholve: keep /nix/store/vw9lf2d0cfw92dsascyv7q1l5w260r2g-jq-1.6-bin/bin/jq
```

</details>

Aside: PATH wrapping is still a fine approach and is even essential for some scripts. resholve does what it can, but Shell is incredibly dynamic/complex.

Comment (urandom2) [reply pkgs/applications/networking/wgnord/default.nix:46]: no, I am a big fan of this solution, as it has the simplixity of `--prefix PATH`, with the effect of `patchelf

File: pkgs/applications/networking/wgnord/default.nix
Comment (urandom2) [inline pkgs/applications/networking/wgnord/default.nix:57]: this works, and despite coming from [a gist](https://gist.github.com/PedroHLC/e6cefa4dce6474d27f3a91f4ff929a67) and feeling like a bodge, seems canonical based on [upstream comments](https://github.com/abathur/resholve/issues/77#issuecomment-1076494874)?

@abathur: can you assist me in the "ideal outcome", and show me how to "improve resholve and/or binlore", as I am keen "to fix it for everyone"?

Comment (abathur) [reply pkgs/applications/networking/wgnord/default.nix:57]: It is, (but I did the same in my diff: https://github.com/abathur/nixpkgs/commit/d6030d4be4a6c8fb084f617d9991b5211405596e#diff-30325ce60c431ba4328f918690dc08e9999399684a0747ba4af0e9648c238bc4R52-R55).

I don't have a great guide on these yet (though I did start one at some point; trying to defer this work until someone's asking so that it can be shaped by feedback).

We can discuss this elsewhere (Matrix? resholve's repo?), but the broad strokes are:
1. If binlore isn't handling the language/format these executables are in, we can add/improve its YARA rules
2. binlore uses very loose heuristics to look for any exec behavior, so if it *is* handling the language/format, we need to look at the source to disambiguate.
    - If it isn't actually user-controlled from the CLI, we can set an override in binlore to ~whitelist it.
    - Since these overrides are heavy-handed/fragile (the executable could gain exec in some future version), it would be _extremely_ ideal to lay down some practices for getting the assumptions from hand audits under test. (No expectation here. Just a task that needs doing. I have some WIP somewhere, but I haven't had time to revisit it for a while.)
3. If it *does* have user-controlled exec behavior, resholve needs a parser that can identify invocations that use it. This is some mix of collecting research and writing a parser.

File: pkgs/applications/networking/wgnord/default.nix
Comment (abathur) [inline pkgs/applications/networking/wgnord/default.nix:54]:
```suggestion
    fix.aliases = true; # curl command in an alias
    execer = [
```

This one was also in my diff. Aliases are a bit unfortunate. When they're in a user's actual shell profile, resolving them generally isn't right. In a script, they often should be. 

I guess it might be safer, if a little pickier, if we flip it around: always check the aliases for commands, and require consent for *either* outcome if commands were found in them.

File: pkgs/applications/networking/wgnord/default.nix
Comment (urandom2) [reply pkgs/applications/networking/wgnord/default.nix:57]: yeah, your call on if you want to spawn a ticket based on this in resholve or hit me up on matrix ([info](https://github.com/NixOS/nixpkgs/blob/d87fcbe3cd948f8f1616723d2ed2e05f13ca13a5/maintainers/maintainer-list.nix#L14561) in the maintainers file); I will resolve this to ensure that there is no confusion about the status of this pr

File: pkgs/applications/networking/wgnord/default.nix
Comment (abathur) [inline pkgs/applications/networking/wgnord/default.nix:37]:
```suggestion
    install -Dm 755 wgnord -t $out/bin/
```
Yeah?

File: pkgs/applications/networking/wgnord/default.nix
Comment (abathur) [inline pkgs/applications/networking/wgnord/default.nix:26]:
```suggestion
```

Ditto :)

File: pkgs/applications/networking/wgnord/default.nix
Comment (abathur) [inline pkgs/applications/networking/wgnord/default.nix:10]:
```suggestion
```

Don't need this anymore :)

**Review by abathur — CHANGES_REQUESTED** [review]
This built fine with `nixpkgs-review` on x86_64-linux (NixOS), butI have a few nits to pick.

File: pkgs/applications/networking/wgnord/default.nix
Comment (urandom2) [reply pkgs/applications/networking/wgnord/default.nix:51]: I tend to treat all applications as black boxes and my goal with this whole process, especially the move to resholve, is to be explicit about dependencies; thus if you want a shell alias to propagate into wgnord, I would expect a user would use a derivation to do it: (or an officially exposed flag or env var by the upstream tool)
```nix
pkgs.wgnord.override { curl = curlWithGnuTls; }
```

Comment (abathur) [reply pkgs/applications/networking/wgnord/default.nix:51]: Not sure if we're on same page. In this case I mean the ones defined in wgnord:

```bash
alias query="curl -s -H 'User-Agent: NordApp Linux 3.12.0 Linux 5.4.0-58-generic'"
```

from https://github.com/phirecc/wgnord/blob/79c4754f9564e6e8e6a3723d2d37d08b67abd2ea/wgnord#L2

So the resholve directive is just saying, "yes, parse aliases to resolve into them".

File: pkgs/applications/networking/wgnord/default.nix
Comment (abathur) [inline pkgs/applications/networking/wgnord/default.nix:41]:
```suggestion
    interpreter = "${bash}/bin/sh";
```

I'm not sure if this one's meaningful, but I did notice that wgnord uses a /bin/sh shebang:
https://github.com/phirecc/wgnord/blob/79c4754f9564e6e8e6a3723d2d37d08b67abd2ea/wgnord#L1

File: pkgs/applications/networking/wgnord/default.nix
Comment (urandom2) [reply pkgs/applications/networking/wgnord/default.nix:51]: yep, I had my wires crossed; and yeah, i like the idea of parsing aliases by default, but happy to do that explicitly

**Review by abathur — APPROVED** [review]

File: pkgs/applications/networking/wgnord/default.nix
Comment (SuperSandro2000) [reply pkgs/applications/networking/wgnord/default.nix:43]: > Can you elaborate on what you mean? Nixos is the one that works ootb, so I am unclear what "add[ing] them to nixos installation" does to resolve the issue, or do you mean the "nix install" should setup PATH to contain everything in stdenv?

It should be expected that any those packages exist on any system and they are the nix version.
Also soon as programs execute other programs like terminals or shells it is impossible to get this right because then the nix version is always preferred. 

----

Wrapping every package in a manual way is definitely not the solution to this. This must be solved globally otherwise we get stuck in a micro managing rabbit hole.

Comment (SuperSandro2000) [reply pkgs/applications/networking/wgnord/default.nix:43]: So only curl, iproute2, jq and wireguard-tools should be in this wrapper.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/networking/wgnord/default.nix
Comment (urandom2) [reply pkgs/applications/networking/wgnord/default.nix:43]: we are no longer wrapping anything; please advise

**Pull request merged**
